sharktank/sharktank/evaluate/perplexity_iree.py

```diff
@@ -215,6 +215,33 @@ def load_model(
 
         self.generator = TorchGenerator(model, tokenizer)
 
+        shard_count = self.tensor_parallelism_size
+
+        vm_instance = ireert.VmInstance()
+        self.devices: list[iree.runtime.HalDevice] = get_iree_devices(
+            device=self.iree_devices,
+            device_count=self.pipeline_parallelism_size * shard_count,
+            allow_repeating=True,
+        )
+        hal_module = iree.runtime.create_hal_module(
+            instance=vm_instance, devices=self.devices
+        )
+        self.weight_path = Path(self.weight_path_str)
+        parameter_index = iree.runtime.ParameterIndex()
+        handles = get_file_handle(shard_count=shard_count, weight_path=self.weight_path)
+        for handle in handles:
+            parameter_index.load_from_file_handle(handle, "irpa")
+
+        parameter_provider = parameter_index.create_provider(scope="model")
+        parameters_module = iree.runtime.create_io_parameters_module(
+            vm_instance, parameter_provider
+        )
+        self.vm_module = iree.runtime.VmModule.mmap(vm_instance, str(self.output_vmfb))
+        self.vm_context = iree.runtime.VmContext(
+            instance=vm_instance,
+            modules=(hal_module, parameters_module, self.vm_module),
+        )
```

Comment:
Is there a reason to not use `load_iree_module` here? If `load_iree_module` does not support something that is required we should add it there.